asm/jump.go

```diff
@@ -62,48 +62,44 @@ func (op JumpOp) Op(source Source) OpCode {
 
 // Imm compares 64 bit dst to 64 bit value (sign extended), and adjusts PC by offset if the condition is fulfilled.
 func (op JumpOp) Imm(dst Register, value int32, label string) Instruction {
-	return Instruction{
-		OpCode:    op.opCode(JumpClass, ImmSource),
-		Dst:       dst,
-		Offset:    -1,
-		Constant:  int64(value),
-		Reference: label,
-	}
+	return (Instruction{
+		OpCode:   op.opCode(JumpClass, ImmSource),
+		Dst:      dst,
+		Offset:   -1,
+		Constant: int64(value),
+	}).SetReference(label)
```

Comment:
The point of the setter is to abstract the metadata struct away so we might change its implementation later. It is not strictly necessary since we are in the same package,  but this might save us refactoring work down the line. I don't feel strongly either way

Reply:
Let's keep it as it is.